watertap/tools/parameter_sweep/parameter_sweep_reader.py

```diff
@@ -90,8 +90,70 @@ def get_sweep_params_from_yaml(self, m, yaml_filename):
         return self._dict_to_params(m, input_dict)
 
     @staticmethod
-    def _dict_to_params(m, input_dict):
+    def _dict_to_diff_spec(m, input_dict):
+        """Reads and stores a yaml file as a dictionary
+        Args:
+            dict (str):
+                The dictionary of paramters that are turned into paramter sweep samples
+        Returns:
+            input_dict (dict):
+                The result of reading the yaml file and translating
+                its structure into a dictionary.
+        """
+        diff_spec = {}
+
+        for param, values in input_dict.items():
+            # Find the specified component on the model
+            component = m.find_component(values["param"])
+
+            if component is None:
+                raise ValueError(f'Could not acccess attribute {values["param"]}')
+
+            if values["diff_sample_type"] == "NormalSample":
+                diff_spec[param] = {
+                    "diff_mode": values["diff_mode"],
+                    "diff_sample_type": NormalSample,
+                    "std_dev": values["std_dev"],
+                    "pyomo_object": component,
+                }
+
+            elif values["diff_sample_type"] == "UniformSample":
+                if values["diff_mode"] == "percentile":
+                    nominal_lb = values["nominal_lb"]
+                    nominal_ub = values["nominal_ub"]
+                else:
+                    nominal_lb = None
+                    nominal_ub = None
+                diff_spec[param] = {
+                    "diff_mode": values["diff_mode"],
+                    "diff_sample_type": UniformSample,
+                    "relative_lb": values["relative_lb"],
+                    "relative_ub": values["relative_ub"],
+                    "nominal_lb": values["nominal_lb"],
+                    "nominal_ub": values["nominal_ub"],
+                    "pyomo_object": component,
+                }
+            elif values["diff_sample_type"] == "LinearSample":
+                if values["diff_mode"] == "percentile":
+                    nominal_lb = values["nominal_lb"]
+                    nominal_ub = values["nominal_ub"]
+                else:
+                    nominal_lb = None
+                    nominal_ub = None
+                diff_spec[param] = {
+                    "diff_mode": values["diff_mode"],
+                    "diff_sample_type": LinearSample,
+                    "relative_lb": values["relative_lb"],
+                    "relative_ub": values["relative_ub"],
+                    "nominal_lb": values["nominal_lb"],
+                    "nominal_ub": values["nominal_ub"],
```

Comment:
Shouldn't these point to `nominal_lb` and `nominal_ub` based on the logic above regarding `diff_mode` being `precentile` or not?

Reply:
Yes fixed that in next commit.